packages/rum-core/src/performance-monitoring/metrics.js

```diff
@@ -317,12 +317,12 @@ export class PerfEntryRecorder {
   }
 
   start(type) {
-    /**
-     * Safari throws an error when PerformanceObserver is
-     * observed for unknown entry types as longtasks and lcp is
-     * not supported at the moment
-     */
     try {
+      const supportedEntryTypes =
+        window.PerformanceObserver?.supportedEntryTypes || []
+      if (supportedEntryTypes.indexOf(type) === -1) {
```

Comment:
Just thinking out loud:

[supportedEntryTypes](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/supportedEntryTypes#browser_compatibility) became available later than [observe](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/observe)

For instance:

                                            Chrome    Safari   Firefox
observe                                   52          11            57
supportedEntryTypes            73           12.1        68

If we stop reporting based on supportedEntryTypes we might start missing data. Okay, **maybe** in the case of Chrome and Firefox it's not a big deal since version 73 and 68 are old. Besides, they are [evergreen](https://learn-the-web.algonquindesign.ca/topics/browser-testing/#evergreen-browsers) browsers.

The case of Safari might be different. We support older versions such the 12. What is more, Safari is not an evergreen browser. 😭  As you can see [here](https://css-tricks.com/evergreen-does-not-mean-immediately-available/#aa-the-browsers-themselves) and [here](https://thingsthemselves.com/reminder-safari-is-not-an-evergreen-browser/)

It seems that we **might** be adding a breaking change discreetly.

Questions:

- Does the firefox warning has more importance than the versioning thingy?
- Is firefox the only browser reporting the warnings?

If firefox is the only one complaining and you both believe that what I commented above might be a problem, a possible proposal might be:

**Step 1:**

We are in 5.x version, we don't want new releases to cause unexpected regressions. What about checking if the user agent is firefox for the following entry types: **largest-contentul-paint,** **layout-shift**, and **long-task**. I know that checking user-agents is tricky. But I wanted to share this at least.

**Step 2:**

Once we go to 6.x version,  there is where we can do breaking changes. For instance, "We support the following browsers, etc, etc". And then we can remove the logic added in **Step 1**


--

I considered that at least was important to bring out this concern

What are your thoughts, @kyungeunni @vigneshshanmugam?

Reply:
This is a great point, thanks for bringing this up.
![image](https://user-images.githubusercontent.com/16660866/174265674-ecea23ba-5723-41de-aacd-dcc124c7aa49.png)

As you mentioned, most of the browsers shipped `supportedEntryTypes` later than `observe` so I can see this change will introduce regression as fewer metrics will be observed, even though the entryType is supported for the browsers.

I appreciate your suggestion on Step1, although would it be fair to assume the possibility each version of FF could have a different set of supported entry types? (e.g. let's say version 57 supports 3 entry types, version 60 supports 5 so not only **largest-contentul-paint**, **layout-shift,** and **long-task** would log the warning messages but other unsupported ones too). Also, there are uninvestigated browsers too that could possibly log the warning messages as same as FF does. That being said, it seems like we can't be sure to cover all the cases with this approach, It'd look like we didn't really fix this bug as we didn't cover all the cases.

I'd love to chat about plans for supported browsers for `6.x` and plan this change accordingly. Meanwhile, could we communicate with users that this is how Firefox-specific behavior and if we try to mute the log, it would introduce regression as we will capture fewer metrics? If it's not a critical bug from the user's perspective, that is.

Reply:
I agree with you here. Likely, the most sensible thing we can do is not to mute the log since it would cause a regression and then wait until 6.x to retackle this. I would like to know @vigneshshanmugam thoughts before closing the PR though.

The issue related to this PR probably will end up having the breaking-change label. And also we will need inform about our decision to the support engineer that brought out this issue.

Reply:
I would like us to think this in a different way. The main types we use for Observer are `largest-contentful-paint`, `long-tasks`, `paint`, `input-delay` - https://github.com/elastic/apm-agent-rum-js/blob/96c7e692c79528e9074408be9d94212f0a7cb375/packages/rum-core/src/performance-monitoring/transaction-service.js#L166-L169 .


 I believe none of those metrics would be supported by the older browsers such as `Safari 11`. If we can confirm that, then we should be good to do just doing a check with `supportedEntryTypes` even in 5.x versions. 

Let me know your thoughts.

Reply:
@vigneshshanmugam thanks for your comment.

It's true, PAINT started to be compatible on Safari 14.1. I wouldn't worried about Chrome either, even if PAINT was available on 60 and supportedEntryTypes on 73. I understand that rely on Chrome being an evergreen browser allows us to be a bit more flexible with the browser support here.

🎉